arrays_strings/permutation/permutation_solution.ipynb

```diff
@@ -101,6 +101,8 @@
     "    def is_permutation(self, str1, str2):\n",
     "        if str1 is None or str2 is None:\n",
     "            return False\n",
+    "        if str1 == ''   or str2 == '':\n",
```

Comment:
Since we are checking for `None` is line 102 and for an empty string in line 104, I think we can do the following check to catch both:

```
        if not str1 or not str2:
            return False
```

Could we also update the "Hash Map Lookup" alternate solution with this check?

Thanks!